python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -148,6 +148,22 @@ def get_lsst_photometric_uncertainties(self):
                                                  'lsstBandpassDict')
 
 
+    def calculateVisibility(self, magFilter, m5, sigma=0.12):
+        """
+        Calculate the probability of detecting a particular source.
+
+        @ param [in] magFilter is the magnitude of the object in the given filter.
+
+        @ param [in] obs_metadata is the observation metadata for a given visit (to determine m5).
+
+        @ param [out] visibility (0/1).
+        """
+        if len(magFilter) == 0:
+            return numpy.array([])
+        completeness = 1.0 / (1 + numpy.exp((magFilter - m5)/sigma))
+        probability = numpy.random.random_sample(len(magFilter))
```

Comment:
The catalogs won't be reproducible if we do this (numpy.random will be seeded from the system time each time the code is run).

Of course... even if we control the seed of numpy.random, we cannot guarantee that SQL will return objects in the same order.

I guess the best thing to do is just add a warning to the docstring explaining that the output of this getter is not going to be repeatable, even for nominally identical catalogs.